### PR TITLE
[#168304371] Completed project configuration

### PR DESCRIPTION
**Pull request by d-al-ibm**
added certs folder
added testenv conf folder
added variables to env file for api
changed value of variable in env file for front end
added command in dockerfile to copy nginx conf file
added custom nginx conf file

### CODE REVIEW

**Comment by cloudify** [comment]
why we need spid testenv2 in this repo that is just static frontend files?

**Comment by d-al-ibm** [comment]
Both folders are needed to test front end locally with docker; these folders can be kept outside of repository, but if someone clone it and needs to run the app locally with spid integration, docker-compose file requires them (volumes creation in docker-compose).
One possible option is to keep these files from back end repository and if the user needs them get them from there.
Another option is to make the user clone both repositories (front end and back end) for and end-to-end test and keep all separate but in these case other changes has to be done (i.e. change docker-compose and other files)
Which is the best way to achieve this (end to end local testing with backend and spid)?

**Comment by cloudify** [comment]
put everything in a single repository? the fact that you need BOTH frontend and backend to develop the app should lead to the obvious conclusion that this is a single application that needs to stay in a single repository

**Comment by d-al-ibm** [comment]
Certs and testenv folders are only needed to test spid integration, not the whole backend. Front end can be tested locally also without the backend with some changes/additions (see point 3 later in this comment) but it seemed more useful and complete to test also with the real backend. About the repository division, I was not the creator of the repositories so I can’t say but maybe at the time it was created it seemed more clean to have them separated and these problems where not clear, also is not uncommon to have a similar setup with separated repository.
Apart from this, the possible solution I see are the following:
1) Insert this folders in the front end project as done in this pull request, but this is redundant and so not viable
2) Keep this folders locally to allow local testing, but this seems a bad and not viable solution as someone who needs to develop and test the front end needs this files too; it can be specified in the readme to go get this folders in backend repository as a quick&dirty solution but again it is far from best practice in my opinion, but it is the fastest solution after point 1
3) Keep everything separated, change the docker compose and remove these folders from the front end. So if someone needs to test everything he has to clone also the backend repository and start both locally; if someone wants just to use clone the front end, something like https://github.com/typicode/json-server or https://github.com/micromata/http-fake-backend (or something similar if something else is preferred) can be included in the front end to mock apis locally so that everything (except spid login probably) can be tested without backend needed 
4) put everything inside a single repository as requested
I don’t know if there are other possible solutions or if apart from solution 4 one of the other is viable. If not, I need to know how to proceed as I think I can’t/shouldn’t create/delete/change repositories in Team Digitale GitHub account.

File: nginx.conf [hunk 1]
```diff
@@ -0,0 +1,58 @@
+#user  nobody;
```
[the remark this answers is not in the document]
Comment (d-al-ibm) [reply nginx.conf]: to work with SPAs, nginx (more generally the https server) needs to be configured to always return index.html if it does not find the specified path http://nginx.org/en/docs/http/ngx_http_core_module.html#try_files. This prevents error in cases like redirecting to a specific page (for example redirecting to /dashboard after spid login) as there is only one index.html and routing is managed by react router

File: package.json [hunk 2]
```diff
@@ -6,7 +6,7 @@
   "author": "https://teamdigitale.governo.it",
   "license": "MIT",
   "scripts": {
-    "start": "chmod +x ./env.sh && ./env.sh && cp env-config.js ./public/ && parcel ./public/index.html --public-url /",
+    "start": "chmod +x ./env.sh && ./env.sh && cp env-config.js ./public/ && rm -rf env-config.js && parcel ./public/index.html --public-url /",
```
[the remark this answers is not in the document]
Comment (d-al-ibm) [reply package.json]: to keep project root clean, i delete the file after copying it to public folder (where it is actually needed)

File: .env.io-onboarding-pa-api.development [hunk 3]
```diff
@@ -4,11 +4,14 @@ PGDATABASE=io_onaboarding_pa_api
 PGPASSWORD=io_onaboarding_pa_api
 
 CLIENT_SPID_ERROR_REDIRECTION_URL=http://localhost:8080/spid-login-error
-CLIENT_SPID_SUCCESS_REDIRECTION_URL=http://localhost:8080/dashboard
+CLIENT_SPID_SUCCESS_REDIRECTION_URL=http://localhost:8080/delegate-dashboard
```
[the remark this answers is not in the document]
Comment (d-al-ibm) [reply .env.io-onboarding-pa-api.development]: the value of this variable needs to be changed in the helm chart too but the variable itself is already present in the helm chart
with further discussion I decided to come back to simply "dashboard" and not "delegate-dashboard" so I will revert this commit and nothing needs to be changed in the helm chart

File: .env.io-onboarding-pa.development [hunk 4]
```diff
@@ -1,2 +1,2 @@
-IO_ONBOARDING_PA_API_HOST=http://io-onboarding-pa-api
+IO_ONBOARDING_PA_API_HOST=http://localhost
```
[the remark this answers is not in the document]
Comment (d-al-ibm) [reply .env.io-onboarding-pa.development]: about the variables in .env.io-onboarding-pa-api.development, these are on backend side, @alexgpeppe said to add them to test but I think he already added them to backend helm chart
about the variables in .env.io-onboarding-pa.development, these are already present in front end helm chart, I only changed the value for local testing with docker

**Pull request closed without merging**